scripts/variantstore/wdl/extract/hail_create_vat_inputs.py

```diff
@@ -73,19 +73,70 @@ def remove_too_many_alt_allele_sites(vds):
 
 
 
+def annotate_adj(
+        mt: hl.MatrixTable,
+        adj_gq: int = 30,
+        adj_ab: float = 0.2,
+) -> hl.MatrixTable:
+    """
+    Annotate genotypes with adj criteria (assumes diploid).
+    Defaults similar to gnomAD values, but GQ >= 20 changed to GQ >= 30 to make up for lack of DP filter.
+    """
+    if "GT" not in mt.entry and "LGT" in mt.entry:
+        print("No GT field found, using LGT instead.")
+        gt_expr = mt.LGT
+    else:
+        gt_expr = mt.GT
+    if "AD" not in mt.entry and "LAD" in mt.entry:
+        print("No AD field found, using LAD instead.")
+        ad_expr = mt.LAD
+    else:
+        ad_expr = mt.AD
+    return mt.annotate_entries(
+        adj=get_adj_expr(
+            gt_expr, mt.GQ, ad_expr, adj_gq, adj_ab
+        )
+    )
+def get_adj_expr(
+    gt_expr: hl.expr.CallExpression,
+    gq_expr: Union[hl.expr.Int32Expression, hl.expr.Int64Expression],
+    ad_expr: hl.expr.ArrayNumericExpression,
+    adj_gq: int = 30,
+    adj_ab: float = 0.2,
+) -> hl.expr.BooleanExpression:
+    """
+    Get adj genotype annotation.
+    Defaults similar to gnomAD values, but GQ >= 20 changed to GQ >= 30 to make up for lack of DP filter.
+    """
+    return (
+        (gq_expr >= adj_gq)
+        & (
+            hl.case()
+            .when(~gt_expr.is_het(), True)
+            .when(gt_expr.is_het_ref(), ad_expr[gt_expr[1]] / hl.sum(ad_expr) >= adj_ab)
+            .default(
+                (ad_expr[gt_expr[0]] / hl.sum(ad_expr) >= adj_ab)
+                & (ad_expr[gt_expr[1]] / hl.sum(ad_expr) >= adj_ab)
+            )
+        )
+    )
```

Comment:
This is probably minor, but Hail is unfortunately currently unable to optimize this to only compute `hl.sum(ad_expr)` once. But if you write it
```suggestion
    total_ad = hl.sum(ad_expr)
    return (
        (gq_expr >= adj_gq)
        & (
            hl.case()
            .when(~gt_expr.is_het(), True)
            .when(gt_expr.is_het_ref(), ad_expr[gt_expr[1]] / total_ad >= adj_ab)
            .default(
                (ad_expr[gt_expr[0]] / total_ad >= adj_ab)
                & (ad_expr[gt_expr[1]] / total_ad >= adj_ab)
            )
        )
    )
```
then we will only compute it once, and we still won't compute it in branches that don't use it.